### PR TITLE
Refactor(eos_designs): Move default platform_settings. network_services_keys and connected_endpoints_keys to schema

### PR DESCRIPTION
**Pull request by gmuloc**
## Change Summary

Currently default platform_settingss are maintained both in schema default values and in the shared_utils python code.
This PR removes the python code.

It also leverages the same mechanism to retrieve the default value for `connected_endpoint_keys` and `network_services_keys` from the schema

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

Read default values from schema

## How to test

Molecule tests are still all passing without changes

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by ClausHolbechArista — COMMENTED** [review]
LGTM. As agreed we will remove the campus comment again.

This should help keep things in sync. We should look into the same for other defaults like network_services_keys and connected_endpoints_keys.

File: python-avd/pyavd/_schema/avdschema.py [hunk 1]
```diff
@@ -155,3 +155,15 @@ def recursive_function(datapath: list, schema: dict) -> dict:
...
         return recursive_function(datapath, schema)
+
+    def get_default_value(self, datapath: list) -> Any:
+        """Return the default value of a key given the datapath as a list.
+
+        Raises:
+        -------
+          AvdSchemaError if no default value is defined.
+        """
+        if "default" not in self.subschema(datapath):
```
Comment (ClausHolbechArista) [inline python-avd/pyavd/_schema/avdschema.py]: for efficiency you should store the subschema in a tmp var.

**Review by JulioPDX — APPROVED** [review]
Great stuff, tested with built-in examples. Configs generated correctly when specifying platforms and connected endpoints keys. LGTM

File: python-avd/pyavd/validate_structured_config.py [hunk 2]
```diff
@@ -24,19 +22,14 @@ def validate_structured_config(structured_config: dict) -> ValidationResult:
...
-    if eos_cli_config_gen_schema_tools is None:
-        eos_cli_config_gen_schema_tools = AvdSchemaTools(schema_id=EOS_CLI_CONFIG_GEN_SCHEMA_ID)
 
     # Inplace conversion of data
-    deprecation_warnings = eos_cli_config_gen_schema_tools.convert_data(structured_config)
+    deprecation_warnings = EosCliConfigGenAvdSchemaTools().convert_data(structured_config)
 
     # Validate input data
-    validation_result = eos_cli_config_gen_schema_tools.validate_data(structured_config)
+    validation_result = EosCliConfigGenAvdSchemaTools().validate_data(structured_config)
```
Comment (ClausHolbechArista) [inline python-avd/pyavd/validate_structured_config.py]:
```suggestion

    eos_cli_config_gen_schema_tools = EosCliConfigGenAvdSchemaTools()

    # Inplace conversion of data
    deprecation_warnings = eos_cli_config_gen_schema_tools.convert_data(structured_config)

    # Validate input data
    validation_result = eos_cli_config_gen_schema_tools.validate_data(structured_config)
```

**Review by ClausHolbechArista — APPROVED** [review]

**Pull request merged**